securedrop/alembic/versions/48a75abc0121_add_seen_tables.py

```diff
@@ -0,0 +1,57 @@
+"""add seen tables
```

Comment:
Still would have preferred last week's version of the tables from the spec, with the addition of the global downloaded/seen flag for dealing with undead unread submissions after journalist deletion, but I'm putting that on the record and putting it down. Moving on with my life. It'll be fine. :slightly_smiling_face:

Reply:
hah! it's not a bad idea. this pr adds a `seen` property to the `submissions` table for dealing with historical data where 'downloaded' was once set by users who may still have accounts or they may not. your idea could work with very little change in addition to what's in this PR if we:

  * started setting 'downloaded' (which would need to be renamed to something that describes that it means 'seen by at least one person') the first time a file/message/reply is seen (i don't think it's necessary to wait until a journalist is deleted to start setting it)
  * added a `downloaded` column (which would have the same brilliant name that indicates that it means 'seen by at least one person') and a `seen` property (similar behavior to the `seen` property on the `submissions` table) to the `replies` table

maybe we'll end up doing this if we decide to cascade delete records in the `seen` tables.

Reply:
We can enable real foreign key handling with the current schema, as you've declared `journalist_id` nullable in the seen tables:
```
sqlite> PRAGMA foreign_keys = ON;
sqlite> insert into seen_replies (reply_id) values (1);
sqlite> select * from seen_replies;
1|1|
```